sklearn/tests/test_pipeline.py

```diff
@@ -393,6 +394,36 @@ def test_pipeline_methods_preprocessing_svm():
         pipe.score(X, y)
 
 
+def test_pipeline_methods_pca_tsne():
+    # test that a only fit_transform needs to be present in order to
+    # run a pipeline with fit_transform.
+    # Don't require transform to be present, explicitly.
+    pca = PCA(n_components=2, random_state=0)
+    tsne = TSNE(random_state=0)
+    separate_emb = tsne.fit_transform(pca.fit_transform(iris.data))
+
+    pca_for_pipeline = PCA(n_components=2, random_state=0)
+    tsne_for_pipeline = TSNE(random_state=0)
+    msg = ("Intermediate step '%s' (type %s) does not have "
+           "transform, pipeline is not reusable." % (tsne_for_pipeline,
+                                                     type(tsne_for_pipeline)))
+    with pytest.warns(UserWarning, match=msg):
+        pipe = make_pipeline(pca_for_pipeline, tsne_for_pipeline)
```

Comment:
The warning is not raised

Reply:
This has been fixed now, it was an issue with the regex match, that should be fixed with explicit strings and callng to `re.escape`.